core/src/convert.rs

```diff
@@ -29,36 +87,212 @@ macro_rules! derive_to_ipld {
     };
 }
 
-macro_rules! derive_to_ipld_generic {
-   ($enum:ident, $ty:ty, $($fn:ident),*) => {
-       impl From<$ty> for Ipld {
-           fn from(t: $ty) -> Self {
-               Ipld::$enum(t$(.$fn())*)
-           }
-       }
-   };
-}
+derive_into_ipld!(Bool, bool, clone);
+derive_into_ipld_prim!(Integer, i8, clone);
+derive_into_ipld_prim!(Integer, i16, clone);
+derive_into_ipld_prim!(Integer, i32, clone);
+derive_into_ipld_prim!(Integer, i64, clone);
+derive_into_ipld_prim!(Integer, i128, clone);
+derive_into_ipld_prim!(Integer, isize, clone);
+derive_into_ipld_prim!(Integer, u8, clone);
+derive_into_ipld_prim!(Integer, u16, clone);
+derive_into_ipld_prim!(Integer, u32, clone);
+derive_into_ipld_prim!(Integer, u64, clone);
+derive_into_ipld_prim!(Integer, usize, clone);
+derive_into_ipld_prim!(Float, f32, clone);
+derive_into_ipld_prim!(Float, f64, clone);
+derive_into_ipld!(String, String, into);
+derive_into_ipld!(String, &str, to_string);
+derive_into_ipld!(Bytes, Box<[u8]>, into_vec);
+derive_into_ipld!(Bytes, Vec<u8>, into);
+derive_into_ipld!(Bytes, &[u8], to_vec);
+derive_into_ipld!(List, Vec<Ipld>, into);
+derive_into_ipld!(Map, BTreeMap<String, Ipld>, to_owned);
+derive_into_ipld!(Link, Cid, clone);
+derive_into_ipld!(Link, &Cid, to_owned);
+
+derive_try_from_ipld!(Bool, bool);
+derive_try_from_ipld!(Integer, i8);
+derive_try_from_ipld!(Integer, i16);
+derive_try_from_ipld!(Integer, i32);
+derive_try_from_ipld!(Integer, i64);
+derive_try_from_ipld!(Integer, i128);
+derive_try_from_ipld!(Integer, isize);
+derive_try_from_ipld!(Integer, u8);
+derive_try_from_ipld!(Integer, u16);
+derive_try_from_ipld!(Integer, u32);
+derive_try_from_ipld!(Integer, u64);
+derive_try_from_ipld!(Integer, u128);
+derive_try_from_ipld!(Integer, usize);
+//derive_from_ipld!(Float, f32);
```

Comment:
What about adding a comment saying something like "f32 conversion is intentionally not done as it would be a lossy tranformation"